### DISCUSSION TITLE
Getting "Error: Cannot find module" when installing custom plugin

### DISCUSSION THREAD

**Discussion by hexcowboy** [NomicFoundation/hardhat#3387] [Troubleshooting] [answered]
We're creating a [hardhat plugin](https://github.com/zerodevapp/hardhat-plugin) for our app. It has a peer dependency [`mock-server`](https://github.com/zerodevapp/mock-server).

I started a new project with `npx hardhat init` to test it out. Here's a link to the reproducible repo ([link](https://github.com/hexcowboy/hardhat-plugin-bug-repro)).

The issue is that when importing both the peer dependency and the plugin in `hardhat.config.ts`, running `npx hardhat` throws an error that the peer dependency cannot be found.

```
$ npx hardhat
An unexpected error occurred:

Error: Cannot find module '@zerodevapp/mock-server'
```

I've tried making the peer dependency a regular npm dependency in the plugin but it also yields the same result. I've tried removing the `package-lock.json` file and `node_modules` folder to reinstall the dependencies. I don't understand why hardhat cannot find the module when the typescript server can.

**Comment by hexcowboy** [comment]
By the way I've found similar issues that were resolved by running `npx hardhat --verbose` to see if a different config was being loaded, but that's not the case here.

**Comment by fvictorio** [comment, marked as answer]
The "Cannot find module" error is correct: if you start a node repl and do `require("@zerodevapp/mock-server")`, you'll get the same error.

I think the problem is that the `pacakge.json` file in that package doesn't have a [main](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#main) entry, so it defaults to `index.js`, which doesn't exist. Try adding `"main": "dist/index.js"` to that package.json.

The reason it did work for `src/index.ts` is (I think) that you are not doing anything with the imports, so no `require`s are generated. This is just a guess though, but it's very likely that's the explanation.

**Reply by hexcowboy** [reply]
thanks, i think you're right. i modified my `mock-server`'s package.json and it appears to be working now
